### PR TITLE
[BE] Feat: search feature

### PR DESCRIPTION
**Pull request by nickparedes**
•	Created an endpoint that takes a word or phrase and returns a list of matching posts.
•	Can be searched by tags
•	The parameters to search for the keyword have to go in the body

Functionality in the application that allows users to search for publications related to specific topics, specific groups, so that the user can find information of interest within the platform.

**API Overview**
The search feature service uses a RESTful API and relies on a POST operation to handle search requests.
The requests and responses with this service are structured in JSON format.

**API Components and Operation**

**1. URL Schema:**
The endpoint to access the search feature service is structured as:
`https://{server}:{port}/search`

**2. Request Body:**
When you send a request to the search feature  service, you must pass the following information in the body:
<html>
<body>

Field | Description | Example
-- | -- | --
keyword| the user passes him the publication he wants to search | Content
labels| the user passes tags to restrict their search| tag1, tag2, tag3

</body>
</html>

tags are passed in a series of strings, for example:
```
“labels” :  [“tag1”,”tag2”,”tag3”]

```

or if you pass it an empty array it will not have any label restrictions
```
“labels” :  []
```
Or simply if we do not put the labels parameter, the service interprets that it is not necessary to search by labels, so the request would look like this:
```
{
    "keyword": "content to search"
}

```

A final example of how our body would look for searching for publications would be like this:
```
{
    "keyword": "content to search",
    "labels": [“tag1”,”tag2”, “tag3”]
}
```
**2. Response Body:**
The response body contains the following structure:
if publications are found:
```
[
	{
		"id_post": 3,
		"id_user": 3,
		"content": " Post content 3",
		"postHour": "2023-09-08 12:00:00",
		"postDate": "2023-09-08",
		"location": " location 3",
		"labels": "tag1,tag2,tag3"
	},
{
		"id_post": 4,
		"id_user": 4,
		"content": " Post content 4",
		"postHour": "2023-09-08 12:00:00",
		"postDate": "2023-09-08",
		"location": " location 4",
		"labels": "tag2,tag3,tag6"
	}
]
```
if there are no matching posts:
```
[
	{
		"message": "No available publications found."
	}
]
```

### CODE REVIEW

**Review by eds000n — COMMENTED** [review]
your PR is labeled as backend change but you have plenty of android changes, please fix it before being able to do any review

File: backend/internal/router.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	r := chi.NewRouter()
+
+	r.Use(middleware.RequestID)
+	r.Use(middleware.Logger)
+
+	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("{\"hello\": \"world\"}"))
+	})
+
+	r.Get("/buscar", func(w http.ResponseWriter, r *http.Request) {
```
Comment (eds000n) [inline backend/internal/router.go:24]: you mean search probably

**Comment by Cristhel17** [comment]
To follow good practices in the project, I leave these recommendations:

- Avoid Deleting .gitignore: It's not advisable to delete the .gitignore file unless there's a strong justification. It helps to keep unwanted files out of the repository.

- Test Coverage: Ensure that the new feature or changes are well-tested to prevent regressions. ( go test )

**Comment by eds000n** [comment]
closed due to lack of response

**Pull request closed without merging**